### ISSUE TITLE
Turn raylib usage into interfaces

### ISSUE THREAD

**Issue by delaneyj** [maxpoletaev/dendy#7]
Is like to use to make examples for my project, https://data-star.dev and NATS.  Basically a live twitch play like thing but without websockets or any user js.  Was going to fork but would you be interested in a PR if successful?

**Comment by maxpoletaev** [comment]
Sounds interesting! Could you please give more details on what exactly you are trying to achieve? 

Getting rid of raylib should be pretty easy, and you don't actually need to abstract it. The emulation itself does not depend on raylib, so in theory, you can "go get" this package and only use its core components to access the frame buffer and input devices but implement your own rendering and input handling on top of that.

That's basically means re-implementing [cmd/dendy/offline.go](https://github.com/maxpoletaev/dendy/blob/main/cmd/dendy/offline.go) for a different rendering backend. A very simplified version of this might be:

```
package main

import (
    "github.com/maxpoletaev/dendy/ines"
    "github.com/maxpoletaev/dendy/input"
    "github.com/maxpoletaev/dendy/system"
)

rom, _ := ines.OpenROM(romFile)
cart, _ := ines.NewCartridge(rom)
joy1 := input.NewJoystick()
nes := system.New(cart, joy1, nil)

for {
    nes.Tick()
    if nes.FrameReady() {
       renderFrame(nes.Frame())
       buttons := readPlayerInput()
       joy1.SetButtons(buttons)
   }
}
```

**Comment by maxpoletaev** [comment]
I actually have had this in mind for a very long time for the wasm renderer, but never got to work on it

**Comment by maxpoletaev** [comment]
Yep, runs fine in a browser https://maxpoletaev.github.io/dendy/

**Comment by delaneyj** [comment]
Awesome thanks!!
